### ISSUE TITLE
Promote ipopt_solver_wrapper to a general utility

### ISSUE THREAD

**Issue by blnicho** [Pyomo/pyomo#728]
In #706 an Ipopt solver wrapper was introduced that reads the Ipopt output and returns a number of useful convergence statistics including the number of iterations, solve time, checking for regularization at the solution, etc.. I think this is a super useful utility that should be moved out of `contrib/parmest` to somewhere like `pyomo/util` or `pyomo/solvers`.

See the wrapper here: https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/parmest/utils/ipopt_solver_wrapper.py

**Comment by carldlaird** [comment]
I agree. This should really be moved to the actual ipopt solver instead of a wrapper function.

**Comment by DLWoodruff** [comment]
It would be best if this functionality was inside ipopt. Second-best would be if the wrapper was implemented as a Pyomo solver.

**Comment by mrmundt** [comment]
Resolved via #3577
